gcc/rust/expand/rust-macro-builtins-asm.cc

```diff
@@ -16,5 +16,528 @@
 // along with GCC; see the file COPYING3.  If not see
 // <http://www.gnu.org/licenses/>.
 
-#include "rust-macro-builtins.h"
-#include "rust-macro-builtins-helpers.h"
+#include "rust-macro-builtins-asm.h"
+
+namespace Rust {
+std::map<AST::InlineAsmOption, std::string> InlineAsmOptionMap{
+  {AST::InlineAsmOption::PURE, "pure"},
+  {AST::InlineAsmOption::NOMEM, "nomem"},
+  {AST::InlineAsmOption::READONLY, "readonly"},
+  {AST::InlineAsmOption::PRESERVES_FLAGS, "preserves_flags"},
+  {AST::InlineAsmOption::NORETURN, "noreturn"},
+  {AST::InlineAsmOption::NOSTACK, "nostack"},
+  {AST::InlineAsmOption::MAY_UNWIND, "may_unwind"},
+  {AST::InlineAsmOption::ATT_SYNTAX, "att_syntax"},
+  {AST::InlineAsmOption::RAW, "raw"},
+};
+
+int
+parseDirSpec (Parser<MacroInvocLexer> &parser, TokenId last_token_id)
+{
+  return 0;
+}
+
+int
+parse_clobber_abi (Parser<MacroInvocLexer> &parser, TokenId last_token_id,
+		   InlineAsmContext &inlineAsmCtx)
```

Comment:
one thing we can look at here is improving error handling by returning a `tl::expected<T, E>` - this way, errors can be properly chained and emitted whenever we'd like